go-controller/pkg/node/gateway_udn.go

```diff
@@ -167,6 +179,32 @@ func (udng *UserDefinedNetworkGateway) AddNetwork() error {
 	if err != nil {
 		return fmt.Errorf("could not add VRF %d for network %s, err: %v", vrfTableId, udng.GetNetworkName(), err)
 	}
+	if udng.openflowManager != nil {
```

Comment:
when do we expect this to be nil? if gateway is disabled? If thats the only case then we should check config.Gateway.Mode and error if it is not disabled && we have no openflowManager? Otherwise couldn't we silently skip programming a network?

Reply:
It was more of a precaution for code moving around places... It will be nil today for localnet topology .. I didn't want a panic in corner cases so checked this.. I can add a else Warning error to this so that we don't silently skip this -> good call out.

Reply:
I'd feel a little better if we checked the positive cases where we know it will be nil, localnet, gateway disabled, etc rather than just warning if nil. But if it something you would rather fix later thats fine.